THRIFT-5570: update java lib document about gradle usage (#2583)
apache/thrift: lib/java/README.md

the Gradle build system, which tends to be predominant amongst Java
developers.

> Currently we use gradle 6.9.2 to build the Thrift Java source. The usual way to setup gradle
> project is to include the gradle-wrapper.jar in the project and then run the gradle wrapper to
> bootstrap setting up gradle binaries. However to avoid putting binary files into the source tree we
> have ignored the gradle wrapper files. You are expected to install it manually, as described in
> the [gradle documentation](https://docs.gradle.org/current/userguide/installation.html), or
> following this step (which is also done in the travis CI docker images):

Reviewer: Minor punctuation, capitalization, and phrasing improvements:

```suggestion
Currently, we use Gradle 6.9.2 to build the Thrift Java source. The usual way to setup Gradle
project is to include the gradle-wrapper.jar in the project and then run the Gradle Wrapper to
bootstrap setting up Gradle binaries. However, to avoid putting binary files into the source tree, we
have excluded the Gradle Wrapper files. You are expected to install Gradle manually, as described in
the [Gradle documentation](https://docs.gradle.org/current/userguide/installation.html), or
following the following steps (which is also done in the Travis CI Docker images):
```

```bash
export GRADLE_VERSION="6.9.2"
# install dependencies
apt-get install -y --no-install-recommends openjdk-11-jdk-headless wget unzip
# download gradle distribution
wget https://services.gradle.org/distributions/gradle-$GRADLE_VERSION-bin.zip -q -O /tmp/gradle-$GRADLE_VERSION-bin.zip
# check binary integrity
echo "8b356fd8702d5ffa2e066ed0be45a023a779bba4dd1a68fd11bc2a6bdc981e8f  /tmp/gradle-$GRADLE_VERSION-bin.zip" | sha256sum -c -
# unzip and install
unzip -d /tmp /tmp/gradle-$GRADLE_VERSION-bin.zip
mv /tmp/gradle-$GRADLE_VERSION /usr/local/gradle
ln -s /usr/local/gradle/bin/gradle /usr/local/bin
```

> After the above step, `gradle` binary will be available in `/usr/local/bin/`. You can further choose
> to locally create the gradle wrapper (even if they are ignored) using:

Reviewer: ```suggestion
After the above step, the `gradle` binary will be available in `/usr/local/bin/`. You may then choose
to locally create the Gradle Wrapper, if you wish to run commands with `gradlew` instead of `gradle`, using:
```

```bash
gradle wrapper --gradle-version $GRADLE_VERSION
```

To compile the Java Thrift libraries, simply do the following:

```bash
gradle
```

> Yep, that's easy. Look for `libthrift-<version>.jar` in the build/libs directory.

Reviewer: We probably shouldn't say it's easy... that's subjective. Users may not agree, and might interpret this as being arrogant.
```suggestion
Look for `libthrift-<version>.jar` in the build/libs directory.
```
